src/Controllers/MaterialCommentController.php

```diff
@@ -51,4 +52,88 @@ public function add(Request $request)
             return ApiResponse::failed($e);
         }
     }
+
+    public function edit(Request $request, $id)
+    {
+        try {
+            $user = auth()->user();
+            $request->validate([
+                'content' => 'required|string|max:65535',
+            ]);
+
+            $materialComment = MaterialComment::where('id', $id)
+                ->where('created_by', $user->id)
+                ->first();
+
+            $lessonMaterial = LessonMaterial::where('id', $materialComment->material_id)
+                ->first();
+
+            if (! $lessonMaterial) {
+                throw ValidationException::withMessages([
+                    'material_id' => 'Lesson Material not found',
+                ]);
+            }
```

Comment:
I think this could be rewritten with the `::with` function. Should be cleaner and you won't need to write the query to fetch lessonMaterial on your own. Please also notice that `$materialComment` could be null, and therefore calling `$materialComment->material_id` will raise a null pointer error